components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/dao/JDBCSubscriberDAOImpl.java

```diff
@@ -0,0 +1,372 @@
+        } finally {
+            IdentityDatabaseUtil.closeConnection(connection);
+        }
+    }
+
+    private void updateSubscriber(String subscriberId, PublisherDataHolder holder, PublisherPropertyDTO[] propertyDTOs,
+                                  PublisherDataHolder oldHolder) throws EntitlementException {
+
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+
+        Connection connection = IdentityDatabaseUtil.getDBConnection(true);
+        try (NamedPreparedStatement updateSubscriberPrepStmt = new NamedPreparedStatement(connection,
+                UPDATE_SUBSCRIBER_MODULE_SQL);
+             NamedPreparedStatement updateSubscriberPropertiesPrepStmt = new NamedPreparedStatement(connection,
+                     UPDATE_SUBSCRIBER_PROPERTIES_SQL)) {
+
+            if (!oldHolder.getModuleName().equalsIgnoreCase(holder.getModuleName())) {
+                updateSubscriberPrepStmt.setString(ENTITLEMENT_MODULE_NAME, holder.getModuleName());
+                updateSubscriberPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                updateSubscriberPrepStmt.setInt(TENANT_ID, tenantId);
+                updateSubscriberPrepStmt.executeUpdate();
+            }
+
+            // Update the property values of an existing subscriber
+            for (PublisherPropertyDTO dto : propertyDTOs) {
+
+                if (StringUtils.isNotBlank(dto.getId()) && StringUtils.isNotBlank(dto.getValue())) {
+                    PublisherPropertyDTO propertyDTO;
+                    propertyDTO = oldHolder.getPropertyDTO(dto.getId());
+                    if (propertyDTO != null && !propertyDTO.getValue().equalsIgnoreCase(dto.getValue())) {
+                        updateSubscriberPropertiesPrepStmt.setString(PROPERTY_VALUE, dto.getValue());
+                        updateSubscriberPropertiesPrepStmt.setString(PROPERTY_ID, dto.getId());
+                        updateSubscriberPropertiesPrepStmt.setString(SUBSCRIBER_ID, subscriberId);
+                        updateSubscriberPropertiesPrepStmt.setInt(TENANT_ID, tenantId);
+                        updateSubscriberPropertiesPrepStmt.addBatch();
+                    }
+                }
+            }
+            updateSubscriberPropertiesPrepStmt.executeBatch();
+            IdentityDatabaseUtil.commitTransaction(connection);
+        } catch (SQLException e) {
+            IdentityDatabaseUtil.rollbackTransaction(connection);
+            throw new EntitlementException("Error while updating subscriber details", e);
+        } finally {
+            IdentityDatabaseUtil.closeConnection(connection);
+        }
+    }
+
+    /**
+     * Sets the base64 encoded secret value of the secret subscriber properties, if it has been updated.
+     *
+     * @param holder    publisher data holder
+     * @param oldHolder existing publisher data holder
+     */
+    private void populateProperties(PublisherDataHolder holder, PublisherDataHolder oldHolder) {
```

Comment:
`populateProperties` -> `encryptUpdatedSecretProperties`

Reply:
Warn: not updated secrets returned in plaintext. check persistent logic whether this is properly handled.

Reply:
> Warn: not updated secrets kept as it is in plaintext. check persistent logic whether this is properly handled.

This is handled in [1] as expected.. But better to avoid having this inter-dependency among method impls.

[1] https://github.com/wso2/carbon-identity-framework/pull/5826/files#diff-50b8fca7333ced1dc9278978245c1128f4d26e7ed4c0449bd944ea1bf6b25608R290